public/js/util/source-map.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+const { networkRequest } = require("./networkRequest");
+const { SourceMapConsumer } = require("source-map");
+
+let sourceMaps = new Map();
+
+function loadSourceMap(tab, source) {
+  const sourceMapURL = `${tab.get("url")}/${source.sourceMapURL}`;
+
+  if (sourceMaps.has(source.id)) {
+    return Promise.resolve(sourceMaps.get(source.id));
+  }
+
+  return networkRequest(sourceMapURL).then(sourceMap => {
+    sourceMaps.set(source.id, sourceMap);
```

Comment:
Why not create the `SourceMapConsumer` here? You don't want to recreate it every time you access it, as it parses the sourcemap. I'd change that before merging this in.